pkg/target/target_test.go

```diff
@@ -430,3 +438,144 @@ func TestProcessData(t *testing.T) {
 		})
 	}
 }
+
+var fooMatch = match.Match{
+	Kinds: []match.Kinds{
+		{
+			Kinds:     []string{"Thing"},
+			APIGroups: []string{"some"},
+		},
+	},
+	Scope:              "Namespaced",
+	Namespaces:         []util.Wildcard{"my-ns"},
+	ExcludedNamespaces: nil,
+	LabelSelector: &metav1.LabelSelector{
+		MatchLabels: map[string]string{
+			"obj": "label",
+		},
+	},
+	NamespaceSelector: &metav1.LabelSelector{
+		MatchLabels: map[string]string{
+			"ns": "label",
+		},
+	},
+	Name: "",
+}
+
+var fooConstraint = makeConstraint(
+	setKinds([]string{"some"}, []string{"Thing"}),
+	setScope("Namespaced"),
+	setNamespaceName("my-ns"),
+	setLabelSelector("obj", "label"),
+	setNamespaceSelector("ns", "label"),
+)
+
+func TestToMatcher(t *testing.T) {
+	unstructAssign, _ := runtime.DefaultUnstructuredConverter.ToUnstructured(&unversioned.Assign{
+		TypeMeta:   metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{},
+		Spec: unversioned.AssignSpec{
+			Match: fooMatch,
+		},
+	})
+	tests := []struct {
+		name       string
+		constraint *unstructured.Unstructured
+		want       constraints.Matcher
+		wantErr    bool
```

Comment:
Use a literal error and check equality with errors.Is() so we're certain we're returning the intended error

Reply:
done.